client/client.go

```diff
@@ -173,7 +173,8 @@ func (c *DatabricksClient) attempt(
 		}
 
 		retry, err := apierr.CheckForRetry(ctx, response, err, responseBody.Bytes(), responseBodyErr)
-		if err != nil && !errors.As(err, &apierr.APIError{}) {
+		var apiErr *apierr.APIError
+		if err != nil && !errors.As(err, &apiErr) {
```

Comment:
This now takes a pointer to a pointer. I don't understand why this is necessary.

Reply:
I'll take a look at the linked issue first.

Reply:
I see the linked documentation that shows defining a pointer type prior to calling `errors.As`.

LGTM, thanks for the PR!